### ISSUE TITLE
Exception: Current Loss Scale Already at Minimum While Fine-Tuning with 8 A800-80G GPUs

### ISSUE THREAD

**Issue by codefanw** [TinyLLaVA/TinyLLaVA_Factory#99]
I encountered the following error while fine-tuning a model using 8 A800-80G GPUs:
`Exception: Current loss scale already at minimum - cannot decrease scale anymore. Exiting run.`
Can anyone help me understand the cause of this issue and how to resolve it?
Looking forward to your prompt response.

**Comment by ggcr** [comment]
This happened to myself when fine-tuning too. It is not a TinyLLaVA problem, it is a Deepspeed problem.

One possible way to fix this is to switch to use `bf16`, as noted in this issue https://github.com/microsoft/DeepSpeedExamples/issues/418. 

Another work-around is to add `fp16_scale_tolerance` into the `scripts/zero3.json` file, which allows to train on `fp16`, which was suggested by Deepspeed in this issue (https://github.com/facebookresearch/fairseq/issues/1529#issuecomment-567955547) I did the latter and had no issue.

```json
{
    "fp16": {
        "enabled": "auto",
        "loss_scale": 0,
        "loss_scale_window": 1000,
        "initial_scale_power": 16,
        "hysteresis": 2,
        "min_loss_scale": 1,
        "fp16_scale_tolerance": 0.25, # add this!
    },
```

I don't quite remember the naming convention followed in zero scripts, I know for sure that the CLI arg is `--fp16-scale-tolerance 0.25`.

**Comment by codefanw** [comment]
I previously tried using bf16 instead of fp16, but in that case, the loss would immediately become NAN. I've tried the second method you suggested, and now I am able to train normally. Thank you very much for your selfless help.
